quint/package.json

```diff
@@ -67,7 +67,7 @@
     "yargs": "^17.2.1"
   },
   "scripts": {
-    "compile": "genversion -e src/version.ts && tsc && copyfiles -u 1 ./src/builtin.qnt ./dist/src/",
+    "compile": "genversion -e src/version.ts && tsc && copyfiles -u 1 ./src/reflection.proto ./src/builtin.qnt ./dist/src/",
```

Comment:
My understanding was that the big benefit of dynamically generating the proto interface rather than generating it at compile time is that we don’t have to add a proto file dep and don’t need a build step. This leads to a ( probably very naive) question: if we are going to be adding a build step to copy over a proto file (which we’ll then use to generate glue code for the reflection I guess) why not instead add a build step the grabs the Apalache proto file? Then we don’t have to worry about the reflection request, and can just read off the interface directly from our proto file. This would let us drop the jar stuff, and and end up with a net reduction in code, rather an increase.

we could even just generate the glue code in that case, saving runtime costs a bit an making it easier to debug.

Am i missing a basic value prop from the reflection approach that justifies the added complexity?

Reply:
Thomas kindly talked this thru with me sync. I'm still not convinced that this is worth the added complexity, given how tightly coupled quint and apalache are. But I am convinced that accepting this change won't cause problems for us (assuming Thomas question is worked out).